### DISCUSSION TITLE
FR: disable/configure rules in VSC settings.json

### DISCUSSION THREAD

**Discussion by lethefrost** [biomejs/biome-vscode#97] [Ideas]
If biome-vscode could read VSC `settings.json` to determine the customized rules when linting in VSC editors, it would be greatly appreciated!

**Comment by sunneydev** [comment]
+1. Biome should be able to be disabled as a whole from `settings.json`.

**Comment by yg-i** [comment]
+1. 

At the same time, can anyone tell me if it's possible at the moment to set Biome to use `{"semicolons": "asNeeded"}` in VSCode, by whatever route? With Prettier you can either use settings UI or the settings.json. Is this possible at all for Biome VSCode?

<img src="https://github.com/biomejs/biome-vscode/assets/148152939/56152fd7-baf2-4b7a-b5a7-56a852dfc65a" width="200" />

**Comment by nemchik** [comment]
+1

I keep ending up disabling the biome extension entirely because many projects I work with do not yet use biome (out of my hands) but I would love to be able to still to use the formatter and just not have the linter showing red all over the editor.

The most desirable behavior for me would be "global" settings for the extension that are ignored when a biome config exists within a project. Ex: I might disable linting entirely from a "global" point of view, but when I open a project with a biome config I would want linting enabled.

**Comment by ezzamo** [comment]
Could anyone confirm whether this is possible yet? it is really annoying to see red all over my code.

**Reply by nhedger** [reply]
This is currently not possible. The current recommended way is to disable the rules in `biome.json`.

**Reply by colinhacks** [reply]
As a +1 I would love the ability to disable Biome in `settings.json`. I'm currently switching often between a `main` and `next` branches of a repo. In `next` we've already migrated to Biome but in `main` we're still using prettier. Currently I have to "Enable/Disable in Workspace" every time I switch because there's no way to indicate persistently that I don't want to use Biome in a given project.

**Reply by nhedger** [reply]
This will be covered by V3. You can track the progress in #201

**Reply by jeremiah-snee-openx** [reply]
@nhedger Was this not included in V3.

This is still happening to me while editing Global/ User `settings.json`

**Reply by nhedger** [reply]
Yes, use `biome.enabled` in the scope in which you want the setting to apply.

https://biomejs.dev/reference/vscode/#biomeenabled

**Comment by Fred-Vatin** [comment]
There is this global settings for user settings.json but it doesn’t work.
![image](https://github.com/user-attachments/assets/4b74bc1a-933e-4a22-b4b1-3132c5e9d9fb)
![image](https://github.com/user-attachments/assets/c919bdb3-98c7-496d-94a0-3e1d986b1864)

I’d like to have a global configuration fallback that could apply to any files opened in VSCode and not only those inside project with a biome.json.

And in project, an init command in palette would be helpful as mentioned at #230.

By the way, I have `"biome_lsp.trace.server": "verbose"` but I can not find any messages in VSCode output.
